src/plugins/interactive_setup/server/routes/views/view_state.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { RouteDefinitionParams } from '..';
+import type { InteractiveSetupViewState } from '../../../common';
+
+/**
+ * Defines routes serving interactive setup mode UI.
+ */
+export function defineViewStateRoutes({
```

Comment:
__note:__ we definitely would need some route that would let UI know the initial/current ES connection status. I used the same approach as I used for Login Selector, but can pick another approach if you have better ideas.

Reply:
What does the UI do with the connection status?

Reply:
I can see these cases:

* If connection status is `not-configured` we should enter into interactive setup mode
* If connection status is `unknown` (user visited Kibana URL before we got response from ES) we may want to show a loading spinner or something like this and poll Kibana server for the status updates. We cannot offer setup yet because it may not be needed if we eventually figure out that default ES connection settings are enough for Kibana to connect to ES.
* If status transitions from `unknown` to `configured`, UI should do exact same thing it does after successful enrollment - wait till Kibana is ready and redirect to originally requested URL
* If status transitions from `unknown` to `not-configured`,  we should enter into interactive setup mode